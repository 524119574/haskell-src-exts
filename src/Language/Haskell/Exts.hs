-----------------------------------------------------------------------------
-- |
-- Module      :  Language.Haskell.Exts
-- Copyright   :  (c) Niklas Broberg 2004-2009
-- License     :  BSD-style (see the file LICENSE.txt)
--
-- Maintainer  :  Niklas Broberg, d00nibro@chalmers.se
-- Stability   :  stable
-- Portability :  portable
--
-- An umbrella module for the various functionality
-- of the package. Also provides some convenient
-- functionality for dealing directly with source files.
--
-----------------------------------------------------------------------------
module Language.Haskell.Exts (
    -- * Re-exported modules
      module Language.Haskell.Exts.Annotated
    ) where

import Language.Haskell.Exts.Annotated

