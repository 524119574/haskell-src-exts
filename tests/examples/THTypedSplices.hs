{-# LANGUAGE TemplateHaskell #-}
import Language.Haskell.TH
main = ($$([|| Just ||]) :: a -> Maybe a) ($$([e|| 2 ||]) :: Int)
