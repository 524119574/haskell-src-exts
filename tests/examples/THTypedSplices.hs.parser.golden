ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 1 1 4 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/THTypedSplices.hs" 1 1 1 1
            , SrcSpan "tests/examples/THTypedSplices.hs" 2 1 2 1
            , SrcSpan "tests/examples/THTypedSplices.hs" 2 1 2 1
            , SrcSpan "tests/examples/THTypedSplices.hs" 2 1 2 1
            , SrcSpan "tests/examples/THTypedSplices.hs" 3 1 3 1
            , SrcSpan "tests/examples/THTypedSplices.hs" 4 1 4 1
            , SrcSpan "tests/examples/THTypedSplices.hs" 4 1 4 1
            ]
        }
      Nothing
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 1 1 1 33
            , srcInfoPoints =
                [ SrcSpan "tests/examples/THTypedSplices.hs" 1 1 1 13
                , SrcSpan "tests/examples/THTypedSplices.hs" 1 30 1 33
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan =
                    SrcSpan "tests/examples/THTypedSplices.hs" 1 14 1 29
                , srcInfoPoints = []
                }
              "TemplateHaskell"
          ]
      ]
      [ ImportDecl
          { importAnn =
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 2 1 2 27
                , srcInfoPoints =
                    [ SrcSpan "tests/examples/THTypedSplices.hs" 2 1 2 7 ]
                }
          , importModule =
              ModuleName
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 2 8 2 27
                  , srcInfoPoints = []
                  }
                "Language.Haskell.TH"
          , importQualified = False
          , importSrc = False
          , importSafe = False
          , importPkg = Nothing
          , importAs = Nothing
          , importSpecs = Nothing
          }
      ]
      [ PatBind
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 1 3 66
            , srcInfoPoints = []
            }
          (PVar
             SrcSpanInfo
               { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 1 3 5
               , srcInfoPoints = []
               }
             (Ident
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 1 3 5
                  , srcInfoPoints = []
                  }
                "main"))
          (UnGuardedRhs
             SrcSpanInfo
               { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 6 3 66
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/THTypedSplices.hs" 3 6 3 7 ]
               }
             (App
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 8 3 66
                  , srcInfoPoints = []
                  }
                (Paren
                   SrcSpanInfo
                     { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 8 3 42
                     , srcInfoPoints =
                         [ SrcSpan "tests/examples/THTypedSplices.hs" 3 8 3 9
                         , SrcSpan "tests/examples/THTypedSplices.hs" 3 41 3 42
                         ]
                     }
                   (ExpTypeSig
                      SrcSpanInfo
                        { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 9 3 41
                        , srcInfoPoints =
                            [ SrcSpan "tests/examples/THTypedSplices.hs" 3 26 3 28 ]
                        }
                      (SpliceExp
                         SrcSpanInfo
                           { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 9 3 25
                           , srcInfoPoints =
                               [ SrcSpan "tests/examples/THTypedSplices.hs" 3 9 3 12
                               , SrcSpan "tests/examples/THTypedSplices.hs" 3 24 3 25
                               ]
                           }
                         (TParenSplice
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/THTypedSplices.hs" 3 9 3 25
                              , srcInfoPoints =
                                  [ SrcSpan "tests/examples/THTypedSplices.hs" 3 9 3 12
                                  , SrcSpan "tests/examples/THTypedSplices.hs" 3 24 3 25
                                  ]
                              }
                            (BracketExp
                               SrcSpanInfo
                                 { srcInfoSpan =
                                     SrcSpan "tests/examples/THTypedSplices.hs" 3 12 3 24
                                 , srcInfoPoints =
                                     [ SrcSpan "tests/examples/THTypedSplices.hs" 3 12 3 15
                                     , SrcSpan "tests/examples/THTypedSplices.hs" 3 21 3 24
                                     ]
                                 }
                               (TExpBracket
                                  SrcSpanInfo
                                    { srcInfoSpan =
                                        SrcSpan "tests/examples/THTypedSplices.hs" 3 12 3 24
                                    , srcInfoPoints =
                                        [ SrcSpan "tests/examples/THTypedSplices.hs" 3 12 3 15
                                        , SrcSpan "tests/examples/THTypedSplices.hs" 3 21 3 24
                                        ]
                                    }
                                  (Con
                                     SrcSpanInfo
                                       { srcInfoSpan =
                                           SrcSpan "tests/examples/THTypedSplices.hs" 3 16 3 20
                                       , srcInfoPoints = []
                                       }
                                     (UnQual
                                        SrcSpanInfo
                                          { srcInfoSpan =
                                              SrcSpan "tests/examples/THTypedSplices.hs" 3 16 3 20
                                          , srcInfoPoints = []
                                          }
                                        (Ident
                                           SrcSpanInfo
                                             { srcInfoSpan =
                                                 SrcSpan
                                                   "tests/examples/THTypedSplices.hs" 3 16 3 20
                                             , srcInfoPoints = []
                                             }
                                           "Just")))))))
                      (TyFun
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/THTypedSplices.hs" 3 29 3 41
                           , srcInfoPoints =
                               [ SrcSpan "tests/examples/THTypedSplices.hs" 3 31 3 33 ]
                           }
                         (TyVar
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/THTypedSplices.hs" 3 29 3 30
                              , srcInfoPoints = []
                              }
                            (Ident
                               SrcSpanInfo
                                 { srcInfoSpan =
                                     SrcSpan "tests/examples/THTypedSplices.hs" 3 29 3 30
                                 , srcInfoPoints = []
                                 }
                               "a"))
                         (TyApp
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/THTypedSplices.hs" 3 34 3 41
                              , srcInfoPoints = []
                              }
                            (TyCon
                               SrcSpanInfo
                                 { srcInfoSpan =
                                     SrcSpan "tests/examples/THTypedSplices.hs" 3 34 3 39
                                 , srcInfoPoints = []
                                 }
                               (UnQual
                                  SrcSpanInfo
                                    { srcInfoSpan =
                                        SrcSpan "tests/examples/THTypedSplices.hs" 3 34 3 39
                                    , srcInfoPoints = []
                                    }
                                  (Ident
                                     SrcSpanInfo
                                       { srcInfoSpan =
                                           SrcSpan "tests/examples/THTypedSplices.hs" 3 34 3 39
                                       , srcInfoPoints = []
                                       }
                                     "Maybe")))
                            (TyVar
                               SrcSpanInfo
                                 { srcInfoSpan =
                                     SrcSpan "tests/examples/THTypedSplices.hs" 3 40 3 41
                                 , srcInfoPoints = []
                                 }
                               (Ident
                                  SrcSpanInfo
                                    { srcInfoSpan =
                                        SrcSpan "tests/examples/THTypedSplices.hs" 3 40 3 41
                                    , srcInfoPoints = []
                                    }
                                  "a"))))))
                (Paren
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/THTypedSplices.hs" 3 43 3 66
                     , srcInfoPoints =
                         [ SrcSpan "tests/examples/THTypedSplices.hs" 3 43 3 44
                         , SrcSpan "tests/examples/THTypedSplices.hs" 3 65 3 66
                         ]
                     }
                   (ExpTypeSig
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/THTypedSplices.hs" 3 44 3 65
                        , srcInfoPoints =
                            [ SrcSpan "tests/examples/THTypedSplices.hs" 3 59 3 61 ]
                        }
                      (SpliceExp
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/THTypedSplices.hs" 3 44 3 58
                           , srcInfoPoints =
                               [ SrcSpan "tests/examples/THTypedSplices.hs" 3 44 3 47
                               , SrcSpan "tests/examples/THTypedSplices.hs" 3 57 3 58
                               ]
                           }
                         (TParenSplice
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/THTypedSplices.hs" 3 44 3 58
                              , srcInfoPoints =
                                  [ SrcSpan "tests/examples/THTypedSplices.hs" 3 44 3 47
                                  , SrcSpan "tests/examples/THTypedSplices.hs" 3 57 3 58
                                  ]
                              }
                            (BracketExp
                               SrcSpanInfo
                                 { srcInfoSpan =
                                     SrcSpan "tests/examples/THTypedSplices.hs" 3 47 3 57
                                 , srcInfoPoints =
                                     [ SrcSpan "tests/examples/THTypedSplices.hs" 3 47 3 51
                                     , SrcSpan "tests/examples/THTypedSplices.hs" 3 54 3 57
                                     ]
                                 }
                               (TExpBracket
                                  SrcSpanInfo
                                    { srcInfoSpan =
                                        SrcSpan "tests/examples/THTypedSplices.hs" 3 47 3 57
                                    , srcInfoPoints =
                                        [ SrcSpan "tests/examples/THTypedSplices.hs" 3 47 3 51
                                        , SrcSpan "tests/examples/THTypedSplices.hs" 3 54 3 57
                                        ]
                                    }
                                  (Lit
                                     SrcSpanInfo
                                       { srcInfoSpan =
                                           SrcSpan "tests/examples/THTypedSplices.hs" 3 52 3 53
                                       , srcInfoPoints = []
                                       }
                                     (Int
                                        SrcSpanInfo
                                          { srcInfoSpan =
                                              SrcSpan "tests/examples/THTypedSplices.hs" 3 52 3 53
                                          , srcInfoPoints = []
                                          }
                                        2
                                        "2"))))))
                      (TyCon
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/THTypedSplices.hs" 3 62 3 65
                           , srcInfoPoints = []
                           }
                         (UnQual
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/THTypedSplices.hs" 3 62 3 65
                              , srcInfoPoints = []
                              }
                            (Ident
                               SrcSpanInfo
                                 { srcInfoSpan =
                                     SrcSpan "tests/examples/THTypedSplices.hs" 3 62 3 65
                                 , srcInfoPoints = []
                                 }
                               "Int")))))))
          Nothing
      ]
  , []
  )
